ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/metering_report_service.proto

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1.admin;
+
+option java_outer_classname = "MeteringReportOuterClass";
+option java_package = "com.daml.ledger.api.v1.admin";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1.Admin";
+
+import "google/protobuf/timestamp.proto";
+
+
+// Experimental API to retrieve metering reports.
+//
+// Metering reports aim to provide the information necessary for billing participant
+// and application operators.
+//
+service MeteringReportService {
+
+    // Retrieve a metering report.
+    rpc GetMeteringReport (GetMeteringReportRequest) returns (GetMeteringReportResponse);
+
+}
+
+// Authorized if and only if the authenticated user is a participant admin.
+message GetMeteringReportRequest {
+
+    // The from timestamp (inclusive).
+    // Required.
+    google.protobuf.Timestamp from = 1;
+
+    // The to timestamp (exclusive).
+    // If not provided, the server will default to its current time.
+    google.protobuf.Timestamp to = 2;
+
+    // If set to a non-empty value, then the report will only be generated for that application.
+    // Optional.
+    string application_id = 3;
+}
+
+message GetMeteringReportResponse {
+
+    // The actual request that was executed.
+    GetMeteringReportRequest request = 1;
+
+    // The computed report.
+    ParticipantMeteringReport participant_report = 2;
+
+    // The time at which the report was computed.
+    google.protobuf.Timestamp report_generated_time = 3;
```

Comment:
Shouldn't that be part of the report itself? 
I am thinking about the generation of that JSON version of that report. If the `from`, `to` and `report_generated_time` were all part of the `participant report`, a grpcurl output could be used "as is".
Throwing it out here for discussion.

Reply:
The response and the report are 1:1 so there was little bit of back and fourth when it came to which is the best field location.  Having it like this means that the report structure should be identical in the scenario where only aggregated data is being used.

Reply:
and I agree that this is a good idea. It's probably really just a question of what we call a "MeteringReport" is the `GetMeteringReportResponse` or a field embedded in there?